fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseTest.java

```diff
@@ -321,10 +319,19 @@ public void testLatencyInjection(FDBLatencySource latencySource, long expectedLa
                 requestedLatency -> requestedLatency == latencySource ? expectedLatency : 0L);
 
         FDBDatabase database = FDBDatabaseFactory.instance().getDatabase();
-        try (FDBRecordContext context = database.openContext()) {
-            long grvStart = System.currentTimeMillis();
-            thingToDo.accept(context);
-            assertTrue(System.currentTimeMillis() - grvStart >= expectedLatency, "latency not injected");
+        try {
+            try (FDBRecordContext context = database.openContext()) {
+                long grvStart = System.currentTimeMillis();
+                thingToDo.accept(context);
+                assertThat("latency not injected without timer", System.currentTimeMillis() - grvStart, greaterThanOrEqualTo(expectedLatency));
+            }
+            FDBStoreTimer timer = new FDBStoreTimer();
+            try (FDBRecordContext context = database.openContext(null, timer)) {
+                long grvStart = System.currentTimeMillis();
+                thingToDo.accept(context);
+                assertThat("latency not injected with timer set", System.currentTimeMillis() - grvStart, greaterThanOrEqualTo(expectedLatency));
```

Comment:
Shouldn't this assertion be different than the one above, or at least the message?

Reply:
I think the assertion is the same. In theory, it is *almost* possible to use the value from the timer to assert on rather than the time here, but there is actually a slight race there in that the timer starts counting right after the latency future is started, so it theoretically could be slightly lower than the target expected latency. (In particular, the code currently (a) gets the injected latency future and then (b) instruments it.) It "shouldn't" be less than, oh, let's say 1 ms different, but I didn't want to add flakiness here by picking the wrong error bars.

As it is, the main reason I added this part of the test (separately from the other section here in this test) is that we've seen instances where the presence or absence of a store timer has changed behavior. I didn't think there was such a bug here, but I figured I'd check.

I think the messages are different already, though ("with" vs "without" timer set).